### PR TITLE
Fancy console badges

### PR DESCRIPTION
**Pull request by nvie**
Adds fancy/branded badges to our console logs:

Screen Shot 2022-09-28 at 22 11 02@2x

These can be used like normal console methods, simply by importing this module as a global:

```tsx
import * as console from "./fancy-console";

console.log('foo');
console.warn('foo');
console.error('foo');
```

---

In addition, there also exists a version that allows you to log out a title (printed in bold):

```tsx
console.errorBold('Some title', 'Lorem ipsum dolor sit amet');
```

Screen Shot 2022-09-28 at 22 16 46@2x

### CODE REVIEW

File: packages/liveblocks-client/src/fancy-console.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          badge,
+          bold,
+          message,
+          ...args
+        );
+}
+
+// export const logBold = wrapBold("log");
+// export const warnBold = wrapBold("warn");
+export const errorBold = wrapBold("error");
```
Comment (nvie) [inline packages/liveblocks-client/src/fancy-console.ts:35]: Not sure about this name. Should we perhaps use `errorWithTitle`?

Comment (FlowFlorent) [reply packages/liveblocks-client/src/fancy-console.ts:35]: I prefer `errorWithTitle` indeed!

Comment (marcbouchenoire) [reply packages/liveblocks-client/src/fancy-console.ts:35]: +1 on `***WithTitle` 👍

File: packages/liveblocks-client/src/fancy-console.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+const badge =
+  "background:radial-gradient(106.94% 108.33% at -10% -5%,#ff1aa3 0,#ff881a 100%);border-radius:6px;color:#fff;padding:2px 6px";
+const bold = "font-weight:600";
```
Comment (nvie) [inline packages/liveblocks-client/src/fancy-console.ts:3]: @pierrelevaillant If we want to change the style of this branded badge, we can simply change the CSS here in one place.

File: packages/liveblocks-client/src/fancy-console.ts (hunk 1 above)
Comment (nvie) [reply packages/liveblocks-client/src/fancy-console.ts]: Thx! Then `xxxWithTItle` it is — f8a98e5493727c62a7a32639f46af2bc65cccbf9!

File: packages/liveblocks-client/.eslintrc.js [hunk 3]
```diff
@@ -6,6 +9,7 @@ module.exports = {
     "@typescript-eslint",
     "eslint-plugin-import",
     "eslint-plugin-simple-import-sort",
+    "eslint-plugin-rulesdir",
```
Comment (nvie) [inline packages/liveblocks-client/.eslintrc.js:12]: The "rulesdir" ESLint plugin allows us to add custom ESLint rules with syntax rules we want to apply to this code base.

File: packages/liveblocks-client/rules/console-must-be-fancy.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      },
+
+      CallExpression(node) {
+        if (node.callee.object?.name === "console" && !seenFancyImport) {
+          context.report({ node: node.callee, messageId: "addMissingImport" });
+        }
+      },
+    };
+  },
+};
```
Comment (nvie) [inline packages/liveblocks-client/rules/console-must-be-fancy.js:26]: Here is a custom ESLint rule I wrote to ensure that we don't accidentally use `console.log()`. Every module that uses a `console.xxx` statement will be reminded about doing that fancy import automatically.

Comment (nvie) [reply packages/liveblocks-client/rules/console-must-be-fancy.js:26]: The reason this needed to be a full custom plugin is that it's a little more complicated rule than just disallowing a certain syntax. We cannot just disallow all console statements. "It depends" on whether the fancy console module has been imported or not. If it were simpler, I wouldn't have used a custom plugin, but instead disallowed the syntax in the .eslintrc file directly, using the `"no-restricted-syntax"` config.

File: packages/liveblocks-client/src/fancy-console.ts (hunk 2 above)
Comment (CTNicholas) [reply packages/liveblocks-client/src/fancy-console.ts:5]: I quite like this styling:
```suggestion
const badge =
  "background:radial-gradient(106.94% 108.33% at -10% -5%,#ff1aa3 0,#ff881a 100%);border-radius:3px;color:#fff;padding:2px 5px;font-family:sans-serif;font-weight:600";
const bold = "font-weight:600";
```

Try it out by pasting this in your console:
```
console.log("%cLiveblocks%c Connection to websocket...", "background:radial-gradient(106.94% 108.33% at -10% -5%,#ff1aa3 0,#ff881a 100%);border-radius:3px;color:#fff;padding:2px 5px;font-family:sans-serif;font-weight:600", "")
```

Comment (nvie) [reply packages/liveblocks-client/src/fancy-console.ts:5]: Nice! Will use that 👍

**Pull request merged**